[Bug] Failed to build BE by GCC-13

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

```shell
ccache /devel/compiler/bin/g++ -DBOOST_STACKTRACE_USE_BACKTRACE -DHAVE_INTTYPES_H -DHAVE_NETINET_IN_H -DUSE_HADOOP_HDFS -I/Programs/doris/be/src/apache-orc/c++/include -I/Programs/doris/be/build_ASAN/src/apache-orc/c++/include -I/Programs/doris/be/build_ASAN/src/clucene/src/shared -I/Programs/doris/be/src/clucene/src/core -I/Programs/doris/be/src/clucene/src/shared -I/Programs/doris/be/src/clucene/src/contribs-lib -I/Programs/doris/be/src -I/Programs/doris/be/test -I/devel/opt/java/include -I/devel/opt/java/include/linux -isystem /Programs/doris/be/../gensrc/build -isystem /Programs/doris/thirdparty/installed/include -isystem /Programs/doris/thirdparty/installed/gperftools/include -O0 -fsanitize=address -DADDRESS_SANITIZER  -std=gnu++20   -D OS_LINUX -g -Wall -Wextra -Werror -pthread -fstrict-aliasing -fno-omit-frame-pointer -Wnon-virtual-dtor -Wno-unused-parameter -Wno-sign-compare -fdiagnostics-color=always -Wno-nonnull -Wno-stringop-overread -Wno-stringop-overflow -D__STDC_FORMAT_MACROS -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX=1 -DBRPC_ENABLE_CPU_PROFILER -DS2_USE_GFLAGS -DS2_USE_GLOG -msse4.2 -mavx2 -DUSE_MEM_TRACKER -DUSE_JEMALLOC -DENABLE_STACKTRACE -gdwarf-5 -MD -MT src/vec/CMakeFiles/Vec.dir/common/format_ip.cpp.o -MF src/vec/CMakeFiles/Vec.dir/common/format_ip.cpp.o.d -o src/vec/CMakeFiles/Vec.dir/common/format_ip.cpp.o -c /Programs/doris/be/src/vec/common/format_ip.cpp
In file included from /Programs/doris/be/src/vec/common/format_ip.cpp:21:
/Programs/doris/be/src/vec/common/format_ip.h:34:19: error: 'array' in namespace 'std' does not name a template type
   34 | extern const std::array<std::pair<const char*, size_t>, 256> one_byte_to_string_lookup_table;
      |                   ^~~~~
/Programs/doris/be/src/vec/common/format_ip.h:26:1: note: 'std::array' is defined in header '<array>'; did you forget to '#include <array>'?
   25 | #include <utility>
  +++ |+#include <array>
   26 |
/Programs/doris/be/src/vec/common/format_ip.h:56:24: error: 'uint8_t' has not been declared
   56 |                        uint8_t mask_tail_octets = 0, const char* mask_string = "xxx") {
      |                        ^~~~~~~
/Programs/doris/be/src/vec/common/format_ip.h: In function 'void doris::vectorized::formatIPv4(const unsigned char*, size_t, char*&, int, const char*)':
/Programs/doris/be/src/vec/common/format_ip.h:66:9: error: 'uint8_t' was not declared in this scope
   66 |         uint8_t value = 0;
      |         ^~~~~~~
/Programs/doris/be/src/vec/common/format_ip.h:26:1: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   25 | #include <utility>
  +++ |+#include <cstdint>
   26 |
/Programs/doris/be/src/vec/common/format_ip.h:68:13: error: 'value' was not declared in this scope
   68 |             value = static_cast<uint8_t>(src[IPV4_BINARY_LENGTH - octet - 1]);
      |             ^~~~~
/Programs/doris/be/src/vec/common/format_ip.h:68:33: error: 'uint8_t' does not name a type
   68 |             value = static_cast<uint8_t>(src[IPV4_BINARY_LENGTH - octet - 1]);
      |                                 ^~~~~~~
/Programs/doris/be/src/vec/common/format_ip.h:68:33: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/Programs/doris/be/src/vec/common/format_ip.h:70:13: error: 'value' was not declared in this scope
   70 |             value = static_cast<uint8_t>(src[octet]);
      |             ^~~~~
/Programs/doris/be/src/vec/common/format_ip.h:70:33: error: 'uint8_t' does not name a type
   70 |             value = static_cast<uint8_t>(src[octet]);
      |                                 ^~~~~~~
/Programs/doris/be/src/vec/common/format_ip.h:70:33: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/Programs/doris/be/src/vec/common/format_ip.h:71:15: error: 'uint8_t' does not name a type
   71 |         const uint8_t len = one_byte_to_string_lookup_table[value].second;
      |               ^~~~~~~
/Programs/doris/be/src/vec/common/format_ip.h:71:15: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/Programs/doris/be/src/vec/common/format_ip.h:72:27: error: 'one_byte_to_string_lookup_table' was not declared in this scope
   72 |         const char* str = one_byte_to_string_lookup_table[value].first;
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Programs/doris/be/src/vec/common/format_ip.h:72:59: error: 'value' was not declared in this scope
   72 |         const char* str = one_byte_to_string_lookup_table[value].first;
      |                                                           ^~~~~
/Programs/doris/be/src/vec/common/format_ip.h:74:26: error: 'len' was not declared in this scope; did you mean 'mblen'?
   74 |         memcpy(dst, str, len);
      |                          ^~~
      |                          mblen
/Programs/doris/be/src/vec/common/format_ip.h: At global scope:
/Programs/doris/be/src/vec/common/format_ip.h:90:62: error: 'uint8_t' has not been declared
   90 | inline void formatIPv4(const unsigned char* src, char*& dst, uint8_t mask_tail_octets = 0,
      |                                                              ^~~~~~~
[4/6] Building CXX object src/vec/CMakeFiles/Vec.dir/functions/function_ip.cpp.o
ninja: build stopped: subcommand failed.
```

### What You Expected?

Build BE successfully.

### How to Reproduce?

Build the codebase by GCC-13

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/src/http/action/pprof_actions.cpp
+++ b/be/src/http/action/pprof_actions.cpp
@@ -42,7 +42,7 @@
 namespace doris {
 
 // pprof default sample time in seconds.
-static const std::string SECOND_KEY = "seconds";
+[[maybe_unused]] static const std::string SECOND_KEY = "seconds";
 static const int kPprofDefaultSampleSecs = 30;
 
 // Protect, only one thread can work

--- a/be/src/runtime/runtime_state.cpp
+++ b/be/src/runtime/runtime_state.cpp
@@ -300,17 +300,11 @@ Status RuntimeState::check_query_state(const std::string& msg) {
     return query_status();
 }
 
-const std::string ERROR_FILE_NAME = "error_log";
 const int64_t MAX_ERROR_NUM = 50;
 
 Status RuntimeState::create_error_log_file() {
     _exec_env->load_path_mgr()->get_load_error_file_name(
             _db_name, _import_label, _fragment_instance_id, &_error_log_file_path);
-    // std::stringstream ss;
-    // ss << load_dir() << "/" << ERROR_FILE_NAME
-    //     << "_" << std::hex << fragment_instance_id().hi
-    //     << "_" << fragment_instance_id().lo;
-    // _error_log_file_path = ss.str();
     std::string error_log_absolute_path =
             _exec_env->load_path_mgr()->get_load_error_absolute_path(_error_log_file_path);
     _error_log_file = new std::ofstream(error_log_absolute_path, std::ifstream::out);

--- a/be/src/util/runtime_profile.cpp
+++ b/be/src/util/runtime_profile.cpp
@@ -38,8 +38,6 @@ namespace doris {
 
 // Thread counters name
 static const std::string THREAD_TOTAL_TIME = "TotalWallClockTime";
-static const std::string THREAD_USER_TIME = "UserTime";
-static const std::string THREAD_SYS_TIME = "SysTime";
 static const std::string THREAD_VOLUNTARY_CONTEXT_SWITCHES = "VoluntaryContextSwitches";
 static const std::string THREAD_INVOLUNTARY_CONTEXT_SWITCHES = "InvoluntaryContextSwitches";
 

--- a/be/src/vec/common/format_ip.h
+++ b/be/src/vec/common/format_ip.h
@@ -19,8 +19,11 @@
 // and modified by Doris
 
 #pragma once
+
 #include <algorithm>
+#include <array>
 #include <bit>
+#include <cstdint>
 #include <cstring>
 #include <utility>
 

